cmd/object-api-common.go

```diff
@@ -309,3 +310,186 @@ func cleanupDir(storage StorageAPI, volume, dirPath string) error {
 	err := delFunc(retainSlash(pathJoin(dirPath)))
 	return err
 }
+
+func checkRange(offsetBegin, offsetEnd int64, resourceSize int64) (startOffset, length int64, err error) {
```

Comment:
No it is on purpose.. httpRange is not the right data structure at object layer and also its not finished yet since its a critical change.  Reviews can be done only after unit tests are added.